File: notes-n-resources/Data-Structures-N-Algo/_DS-n-Algos/__MY_OPRIGINAL_DS/_Extra-Practice/03_recursion/kotlin/02_greet.kt
private fun greet2(name: String) = println("how are you, $name?")

private fun bye() = println("ok bye!")

private fun greet(name: String) {
    println("hello, $name!")
    greet2(name)
    println("getting ready to say bye...")
    bye()
}

fun main(args: Array<String>) = greet("adit")

File: notes-n-resources/Data-Structures-N-Algo/_DS-n-Algos/__MY_OPRIGINAL_DS/_Extra-Practice/03_recursion/javascript/02_greet.js
/**
 * Displays a message to the console
 * @param {string} name Name
 */
function greet2(name) {
  console.log("how are you, " + name + "?");
}

/**
 * Displays a message to the console
 */
function bye() {
  console.log("ok bye!");
}

/**
 * Displays a message to the console
 * @param {string} name Name
 */
function greet(name) {
  console.log("hello, " + name + "!");
  greet2(name);
  console.log("getting ready to say bye...");
  bye();
}

greet("adit");
